docs/dgfit/plot_results.rst

```diff
@@ -15,6 +15,271 @@ For example, to plot the default run results from the optimizer (= max prob):
 
-    plot_dgfit dgfit_test_WD_best_optimizer.fits obsdata --start
\ No newline at end of file
+    plot_dgfit dgfit_test_WD_best_optimizer.fits obsdata --start
+
+To plot the data of the used dustgrains (default is astro-silicates), use command
+
+    plot_dustgrains
+
+.. plot::
+
+    from dgfit.dustgrains import DustGrains
+    import numpy as np
+    import matplotlib.pyplot as pyplot
+    import matplotlib
+    import colorsys
+
+    DG = DustGrains()
+    DG.from_files("astro-silicates", "../../dgfit/data/indiv_grain/")
+
+    fontsize = 12
+    font = {"size": fontsize}
+
+    matplotlib.rc("font", **font)
+
+    matplotlib.rc("lines", linewidth=2)
+    matplotlib.rc("axes", linewidth=2)
+    matplotlib.rc("xtick.major", width=2)
+    matplotlib.rc("ytick.major", width=2)
+
+    fig, ax = plt.subplots(ncols=3, nrows=2, figsize=(15, 10))
+
+    ws_indxs = np.argsort(DG.wavelengths)
+    ews_indxs = np.argsort(DG.wavelengths_emission)
+    waves = DG.wavelengths[ws_indxs]
+    colors = []
+    labels = []
+    for i in range(DG.n_sizes):
+        pcolor = colorsys.hsv_to_rgb(float(i) / DG.n_sizes / (1.1), 1, 1)
+        colors.append(pcolor)
+        labels.append(DG.sizes[i])
+
+        ax[0, 0].plot(waves, DG.cabs[i, ws_indxs], color=pcolor)
+        ax[0, 0].set_xlabel(r"$\lambda$ [$\mu m$]")
+        ax[0, 0].set_ylabel("C(abs)")
+        ax[0, 0].set_xscale("log")
+        ax[0, 0].set_yscale("log")
+
+        ax[0, 1].plot(waves, DG.csca[i, ws_indxs], color=pcolor)
+        ax[0, 1].set_xlabel(r"$\lambda$ [$\mu m$]")
+        ax[0, 1].set_ylabel("C(sca)")
+        ax[0, 1].set_xscale("log")
+        ax[0, 1].set_yscale("log")
+
+        ax[0, 2].plot(waves, DG.cext[i, ws_indxs], color=pcolor)
+        ax[0, 2].set_xlabel(r"$\lambda$ [$\mu m$]")
+        ax[0, 2].set_ylabel("C(ext)")
+        ax[0, 2].set_xscale("log")
+        ax[0, 2].set_yscale("log")
+
+        ax[1, 0].plot(
+            DG.wavelengths_scat_a,
+            DG.scat_a_csca[i, :] / DG.scat_a_cext[i, :],
+            "o",
+            color=pcolor,
+        )
+        ax[1, 0].set_xlabel(r"$\lambda$ [$\mu m$]")
+        ax[1, 0].set_ylabel("albedo")
+        ax[1, 0].set_xscale("log")
+
+        ax[1, 1].plot(DG.wavelengths_scat_g, DG.scat_g[i, :], "o", color=pcolor)
+        ax[1, 1].set_xlabel(r"$\lambda$ [$\mu m$]")
+        ax[1, 1].set_ylabel("g")
+        ax[1, 1].set_xscale("log")
+
+        ax[1, 2].plot(
+            DG.wavelengths_emission[ews_indxs], DG.emission[i, ews_indxs], color=pcolor
+        )
+        ax[1, 2].set_xlabel(r"$\lambda$ [$\mu m$]")
+        ax[1, 2].set_ylabel("Emission")
+        ax[1, 2].set_xscale("log")
+        ax[1, 2].set_yscale("log")
+        cur_ylim = ax[1, 2].get_ylim()
+        ax[1, 2].set_ylim([1e-23, 1e-0])
+
+    ax[0, 1].set_title("astro-silicates")
+    fig.legend(labels, title="Grainsizes [$m$]", loc="lower center", bbox_to_anchor=(0.5, 0), ncol=DG.n_sizes/3)
+
+
+    plt.tight_layout(rect=[0, 0.14, 1, 1])
```

Comment:
This doesn't use the plot_dustgrains code, but replicates it here, right?   It would be better if it directly used the same code that plot_dustgrains uses so we do not have to maintain two versions - one in the code and one in the docs.

Reply:
Yes, I tried to do that but something went wrong. That was when I just started and I did it like this for now. I can try again, with what I know now, it should be easier.